docs/flow/modin/config.rst

```diff
@@ -56,3 +56,21 @@ API.
     # Changing value of `NPartitions`
     modin.config.NPartitions.put(16)
     print(modin.config.NPartitions.get()) # prints '16'
+
+One can also use config variables with a context manager in order to use
+some config only for a certain part of the code:
+
+.. code-block:: python
+
+    import modin.config as cfg
+
+    # Default value for this config is 'False'
+    print(cfg.RangePartitioning.get()) # False
+
+    # Set the config to 'True' inside of the context-manager
+    with cfg.RangePartitioning(True):
```

Comment:
> From the user’s perspective, this is quite convenient

I believe that's the most important thing - be convenient for a user.

> is it really worth adding the ability to create instances from classes that were originally designed without such use case

I also didn't like it at the beginning, but then couldn't find a case where it acts badly, so I think it's okay to add this, what do you think?

Reply:
Without creating the instances, you would need a stack for saving/restoring the previous value, because in case of nested context managers you could miss the original value. Example:
```python
def foo():
    with cfg.RangePartitioning(True):
        print(cfg.RangePartitioning.get())

def bar():
    with cfg.RangePartitioning(False):
        foo()
    print(cfg.RangePartitioning.get())
```

Reply:
> Without creating the instances, you would need a stack for saving/restoring the previous value, because in case of nested context managers you could miss the original value. Example:
> 
> ```python
> def foo():
>     with cfg.RangePartitioning(True):
>         print(cfg.RangePartitioning.get())
> 
> def bar():
>     with cfg.RangePartitioning(False):
>         foo()
>     print(cfg.RangePartitioning.get())
> ```

It is, so I suggest using a separate function:
```python
@contextlib.contextmanager
def update(config, new_value):
    old_value = config.get()
    config.put(new_value)
    try:
        yield
    finally:
        config.put(old_value)


### USING ###
# Set the config to 'True' inside of the context-manager
with cfg.update(cfg.RangePartitioning, True):
    print(cfg.RangePartitioning.get()) # True
    df.merge(...) # will use range-partitioning impl
```

> I also didn't like it at the beginning, but then couldn't find a case where it acts badly, so I think it's okay to add this, what do you think?

@dchigarev I have doubts and there is an alternative. However, if you are confident, then I rely on you.

Reply:
I think, it could be useful to have support for multiple entries:
```python
@contextlib.contextmanager
def context(*ctx):
    configs = ctx[0::2]
    old_values = [config.get() for config in configs]
    for config, value in zip(configs, ctx[1::2]):
        config.put(value)
    try:
        yield
    finally:
        for config, value in zip(configs, old_values):
            config.put(value)

with cfg.context(cfg.RangePartitioning, True, cfg.NPartitions, 1, cfg.MinPartitionSize, 64):
    print(cfg.RangePartitioning().get())
    print(cfg.NPartitions().get())
    print(cfg.MinPartitionSize().get())
```

Reply:
I'll update the PR then using a separate context manager function that supports setting values for multiple configs

Reply:
changed the PR, so now the config change is done via a separate function